tscached/utils.py

```diff
@@ -30,10 +35,19 @@ def get_timedelta(value):
 
 
 def query_kairos(kairos_host, kairos_port, query):
-    """ do it. """
-    url = 'http://%s:%s/api/v1/datapoints/query' % (kairos_host, kairos_port)
-    r = requests.post(url, data=json.dumps(query))
-    return json.loads(r.text)
+    """ As the name states.
+        kairos_host: str, host/fqdn of kairos server. commonly a load balancer.
+        kairos_port: int, port that kairos (or a proxy) listens on.
+        query: dict to send to kairos.
+        returns: dict containing kairos' response.
+        raises: BackendQueryFailure if the operation doesn't succeed.
+    """
+    try:
+        url = 'http://%s:%s/api/v1/datapoints/query' % (kairos_host, kairos_port)
+        r = requests.post(url, data=json.dumps(query))
+        return json.loads(r.text)
+    except requests.exceptions.RequestException as e:
```

Comment:
If it is the case that "requests.exceptions.RequestException" is only raised for failure to connect, would you want to capture any non-2XX response from Kairos here and raise the backend exception?